### PR TITLE
jetcd-core: remove testNG, introduce "bytesOf"

### LINKED ISSUE #309: Cleanup tests : remove TestNG
We have some tests using TestNG an some others using JUnit, we need to get rid of TestNG

### PR DESCRIPTION
**Pull request by yb172**
TestNG is removed to unify stack used for testing, `ByteSequence.from(<str>, UTF_8)` is replaced with more concise `bytesOf(<str>)`

Fixes #309

### CODE REVIEW

**Comment by yb172** [comment]
@fanminshi @lburgazzoli Please take a look

File: jetcd-core/src/test/java/io/etcd/jetcd/AuthClientTest.java [hunk 1]
```diff
@@ -15,262 +15,158 @@
...
+
 import java.net.URI;
 import java.util.List;
-import java.util.concurrent.ExecutionException;
-import org.testng.annotations.AfterTest;
-import org.testng.annotations.BeforeTest;
-import org.testng.annotations.Test;
+
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
```
Comment (fanminshi) [inline jetcd-core/src/test/java/io/etcd/jetcd/AuthClientTest.java]: beforeClass and Classrule is a junit 4 thing right? I think we currently want to move to junit 5. So maybe this refactoring can leverage junit 5 syntax?

This test https://github.com/etcd-io/jetcd/blob/master/jetcd-core/src/test/java/io/etcd/jetcd/KVNamespaceTest.java uses junit 5. Maybe you can base it off that or come with something better>

**Comment by fanminshi** [comment]
overall the pr looks good in general.

File: jetcd-core/src/test/java/io/etcd/jetcd/AuthClientTest.java (hunk 1 above)
Comment (yb172) [reply jetcd-core/src/test/java/io/etcd/jetcd/AuthClientTest.java]: It is JUnit4 thing and I tried to refactor everything to JUnit5 but wasn't able to do so. I haven't found an easy way to convert `@Rule` to JUnit5 extension. As you mentioned before this PR already does 2 things so I didn't want to add one more.

File: jetcd-core/src/test/java/io/etcd/jetcd/AuthClientTest.java [hunk 2]
```diff
@@ -15,262 +15,158 @@
...
-import java.io.IOException;
+import io.etcd.jetcd.launcher.junit.EtcdClusterResource;
+
 import java.net.URI;
 import java.util.List;
-import java.util.concurrent.ExecutionException;
-import org.testng.annotations.AfterTest;
-import org.testng.annotations.BeforeTest;
-import org.testng.annotations.Test;
+
```
[the remark this answers is not in the document]
Comment (yb172) [reply jetcd-core/src/test/java/io/etcd/jetcd/AuthClientTest.java]: I would vote for whatever is supported by most of IDEs by default. For some reason if I tick "optimize on the fly" in IDEA it messes up imports so checkstyle complains.
Let's consider this in a separate issue.

File: jetcd-core/src/test/java/io/etcd/jetcd/AuthClientTest.java (hunk 1 above)
Comment (fanminshi) [reply jetcd-core/src/test/java/io/etcd/jetcd/AuthClientTest.java]: fair enough, would you like to convert this Junit 5 in another pr then?

**Comment by fanminshi** [comment]
lgtm defer to @lburgazzoli

File: jetcd-core/src/test/java/io/etcd/jetcd/AuthClientTest.java (hunk 1 above)
Comment (yb172) [reply jetcd-core/src/test/java/io/etcd/jetcd/AuthClientTest.java]: I would take another look on how to replace `@Rule`. If I would find a way - yes, I'll do another PR, if not - then I won't :)

**Pull request merged**